### PR TITLE
Add io.github.glaumar.QRookie

### PR DESCRIPTION
**Pull request by glaumar**
### Please confirm your submission meets all the criteria

- [x] QRookie downloads and installs Quest games from ROOKIE Public Mirror.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.github.glaumar.QRookie.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/glaumar/QRookie/archive/refs/tags/v0.1.3.tar.gz
+        sha256: 219efb8bbc69d0a408c28792e2a2684f014427ed26e4b346d17fcb5cf9a43572
+
+  - name: cleanup
+    buildsystem: simple
+    build-commands:
+      - rm -rf "$FLATPAK_DEST/lib/librclone.a"
+      - rm -rf "$FLATPAK_DEST/include/librclone.h"
```
Comment (bbhtt) [inline io.github.glaumar.QRookie.yml:59]: Use

```yaml
cleanup:
  - "*.a"
  - /include
```

on top

**Comment by glaumar** [comment]
> > QRookie downloads and installs Quest games from ROOKIE Public Mirror.
> 
> What is the download source and license of these games and is this legal?

Download from https://theapp.vrrookie.xyz/, which is a public server. I think there may be pirated games uploaded by users on it. But QRookie itself has nothing to do with the website, it is just a downloader

File: io.github.glaumar.QRookie.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        sha256: b7c8f00273ad27d85814bf4ec93eb6922c75656800a61d11854d36355a4a1aec
+
+  - name: librclone
+    buildsystem: simple
+    build-commands:
+      - install -pDm 0644 librclone.a -t "$FLATPAK_DEST/lib/"
+      - install -pDm 0644 librclone.h -t "$FLATPAK_DEST/include/"
+    sources:
+      - type: archive
+        url: https://github.com/glaumar/QRookie/releases/download/v0.1.3/librclone_x86_1.65.tar.gz
```
Comment (hfiguiere) [inline io.github.glaumar.QRookie.yml:45]: why not building this from source?

Comment (glaumar) [reply io.github.glaumar.QRookie.yml:48]: When I  try : flatpak-go-get-generator.py .flatpak-builder/build/librclone/ven
dor/modules.txt >  go-sources.json
I get error: 
 Failed to check https://goftp.io/server/v2: HTTP Error 403: Forbidden

File: io.github.glaumar.QRookie.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  - "*.a"
+  - /include
+modules:
+  - name: p7zip
+    buildsystem: simple
+    build-commands:
+      - install -pDm 0755 7za -t "$FLATPAK_DEST/bin/"
+    sources:
+      - type: archive
+        url: https://github.com/p7zip-project/p7zip/releases/download/v17.05/linux-cmake-p7zip.zip
```
[the remark this answers is not in the document]
Comment (glaumar) [reply io.github.glaumar.QRookie.yml:22]: Ok, I will do  it.

[the remark this answers is not in the document]
Comment (glaumar) [reply io.github.glaumar.QRookie.yml]: Sorry to keep you waiting, now build p7zip from source

**Comment by glaumar** [comment]
After updating to v0.2.0, QRookie no longer relies on librclone, and now everything is build from source (except adb)

**Comment by bbhtt** [comment]
We need to discuss this a bit, given the current situation with yuzu etc. please wait.

File: flathub.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (bbhtt) [inline flathub.json:2]: Why is this still x86_64 only?

Comment (glaumar) [reply flathub.json:2]: I have not compiled and tested QRookie under other architectures, and I am not sure whether other dependencies will work under other architectures.

**Comment by glaumar** [comment]
> We need to discuss this a bit, given the current situation with yuzu etc. please wait.
> 
> The website above is also blocked for me by cloudflare, so I can't check what content is there.
Downloading through the browser is a bit cumbersome, it filters UserAgent (such as rclone/v1.65.2), which is why I wrote QRookie.

Downloading through the browser is a bit cumbersome, it filters UserAgent (such as rclone/v1.65.2), which is why I wrote QRookie.

If you want to check the contents, I compiled the flatpak installation package of QRookie at https://github.com/glaumar/QRookie/releases.

If you are a Windows user, you can use https://github.com/VRPirates/rookie to download (rookie is their official application).

I'm pretty sure there are pirated VR games on it, and I can understand if it can't be put on flathub because of that.

Thanks

**Comment by bbhtt** [comment]
Are you leaking some kind of password/token for the site in the output? I see a line with password=XXXXX on terminal.

Also there is a bunch of warnings like `qrc:/qt-project.org/imports/QtQuick/Controls/Material/Button.qml: QML IconImage: Cannot open: qrc:/qt/qml/content/download
qrc:/qt-project.org/imports/QtQuick/Controls/Material/Button.qml: QML IconImage: Cannot open: qrc:/qt/qml/content/delete`

And there is something wrong with theming (this is on Fedora/GNOME 45)

image

A `--env=XDG_CURRENT_DESKTOP=KDE` should resolve it.

**Comment by glaumar** [comment]
> --env=XDG_CURRENT_DESKTOP=KDE

Password is public.
downloaded data on  ~/.var/app/io.github.glaumar.QRookie/data/QRookie/

**Comment by bbhtt** [comment]
> A `--env=XDG_CURRENT_DESKTOP=KDE` should resolve it.

This should also work, if you do https://api.kde.org/frameworks/qqc2-desktop-style/html/index.html `QQuickStyle::setStyle("org.kde.desktop");`

**Comment by glaumar** [comment]
> > > A `--env=XDG_CURRENT_DESKTOP=KDE` should resolve it.
> > 
> > 
> > This should also work, if you do https://api.kde.org/frameworks/qqc2-desktop-style/html/index.html `QQuickStyle::setStyle("org.kde.desktop");`
> 
> Thanks, I will try it

The effect of `QQuickStyle::setStyle(QStringLiteral("org.kde.breeze"))` and `QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"))` on SteamDeck is not what I expected, so I will use it temporarily  `--env=XDG_CURRENT_DESKTOP=KDE`

**Comment by bbhtt** [comment]
bot, build io.github.glaumar.QRookie

**Pull request closed without merging**